src/Algebra/Module/Morphism/Linear/Properties.agda

```diff
@@ -0,0 +1,127 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Properties of linear maps.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra                   using (CommutativeRing)
+open import Algebra.Module            using (Module)
+open import Level                     using (Level)
+
+module Algebra.Module.Morphism.Linear.Properties
+  {r ℓr m ℓm : Level}
+  {ring      : CommutativeRing r ℓr}
+  (modA modB  : Module ring m ℓm)
+  where
+
+import Algebra.Module.Properties            as ModuleProperties
+import Algebra.Module.Morphism.Structures   as MorphismStructures
+
+open import Axiom.DoubleNegationElimination
+open import Data.Product
+open import Relation.Binary.Reasoning.MultiSetoid
+open import Relation.Nullary          using (¬_)
+open import Relation.Nullary.Negation using (contraposition)
+
+open MorphismStructures.ModuleMorphisms modA modB
+module A = Module modA
+open A using () renaming (Carrierᴹ to A)
+module B = Module modB
+open B using () renaming (Carrierᴹ to B)
+module PA = ModuleProperties modA
+module PB = ModuleProperties modB
+open CommutativeRing ring renaming (Carrier to S)
+
+module _
+  {f : A → B}
+  (isModuleHomomorphism : IsModuleHomomorphism f)
```

Comment:
So it's important that each merged in PR stands on its own two legs, in case hypothetical latter PRs don't get merged in.
So in the interests of merging in _this_ PR, again could we make this change now and then we can change it if required in #1781?

Reply:
Ah so you can either write:
```agda
module Algebra.Module.Morphism.ModuleHomomorphism
  {r ℓr m ℓm : Level}
  {ring      : CommutativeRing r ℓr}
  (modA modB  : Module ring m ℓm)
  {f : Module.Carrierᴹ modA → Module.Carrierᴹ modB}
  (isModuleHomomorphism : IsModuleHomomorphism f)
  where
```
or you can do the following neat trick:
```agda
module Algebra.Module.Morphism.ModuleHomomorphism
  {r ℓr m ℓm : Level}
  {ring      : CommutativeRing r ℓr}
  (modA modB  : Module ring m ℓm)
  (open Module modA using () renaming (Carrierᴹ to A))
  (open Module modB using () renaming (Carrierᴹ to B))
  {f : A → B}
  (isModuleHomomorphism : IsModuleHomomorphism f)
  where
```

Reply:
Code changed, as requested, in next push.